pkg/template/api/validation/validation.go

```diff
@@ -61,7 +61,7 @@ func ValidateTemplateInstance(templateInstance *api.TemplateInstance) (allErrs f
 		err.Field = "spec.template." + err.Field
 		allErrs = append(allErrs, err)
 	}
-	if templateInstance.Spec.Secret.Name != "" {
+	if templateInstance.Spec.Secret != nil && templateInstance.Spec.Secret.Name != "" {
```

Comment:
think you need to return a field.Required if Spec.Secret != nil && Spec.Secret.Name == ""